tests/TestLeibnitz.elm

```diff
@@ -0,0 +1,105 @@
+                        (T (PredAtom "p" [Fun "f"[Var "b", Var "b"]] ))
+                        zipperExample
+                    ) 
+                    (Ok (T (PredAtom "p" [Fun "f"[Var "[]", Var "b"]] )))
+            )
+        , test "no change" 
+            (\() ->
+                Expect.equal
+                    (Validation.Rules.Leibnitz.replaceTermWithVar (Fun "f"[Var "b"]) (Var "[]")
+                        (T (Conj (EqAtom (Fun "f"[Var "b"]) (Var "x")) (EqAtom (Fun "f"[Var "b"]) (Var "b")) ) )
+                        (T (Conj (EqAtom (Fun "f"[Var "b"]) (Var "x")) (EqAtom (Fun "f"[Var "b"]) (Var "b")) ) )
+                        zipperExample
+                    ) 
+                    (Ok (T (Conj (EqAtom (Fun "f"[Var "b"]) (Var "x")) (EqAtom (Fun "f"[Var "b"]) (Var "b")) ) )) 
+            )
+        , test "replace in conjunction" 
+            (\() ->
+                Expect.equal
+                    (Validation.Rules.Leibnitz.replaceTermWithVar (Var "c") (Var "[]")
+                        (T (Conj (EqAtom (Fun "f"[Var "b", Var "c"]) (Var "x")) (EqAtom (Fun "f"[Var "c"]) (Var "b")) ) )
+                        (T (Conj (EqAtom (Fun "f"[Var "b", Var "x"]) (Var "x")) (EqAtom (Fun "f"[Var "x"]) (Var "x")) ) )
+                        zipperExample
+                    ) 
+                    (Ok (T (Conj (EqAtom (Fun "f"[Var "b", Var "[]"]) (Var "x")) (EqAtom (Fun "f"[Var "[]"]) (Var "b")) ) )) 
+            )
+        , test "replace in for all" 
+            (\() ->
+                Expect.equal
+                    (Validation.Rules.Leibnitz.replaceTermWithVar (Var "c") (Var "[]")
+                        (T (ForAll "x" (Disj (EqAtom (Fun "f"[Var "x", Var "z"]) (Var "y")) (PredAtom "p" [Fun "f"[Var "a", Var "b", Var "c"], Fun "g"[Var "a"] ]))))
+                       (T (ForAll "x" (Disj (EqAtom (Fun "f"[Var "x", Var "a"]) (Var "y")) (PredAtom "p" [Fun "f"[Var "o", Var "o", Var "o"], Fun "h"[Var "a"] ]))))
+                        zipperExample
+                    ) 
+                    (Ok (T (ForAll "x" (Disj (EqAtom (Fun "f"[Var "x", Var "z"]) (Var "y")) (PredAtom "p" [Fun "f"[Var "a", Var "b", Var "[]"], Fun "g"[Var "a"] ]))))) 
+            )
+        , test "replace in exists" 
+            (\() ->
+                Expect.equal
+                    (Validation.Rules.Leibnitz.replaceTermWithVar (Fun "f"[Var "c", Var "z"]) (Var "[]")
+                        (T (Exists "x" (Conj (EqAtom (Fun "f"[Var "c", Var "z"]) (Var "y")) (PredAtom "p" [Fun "f"[Var "c", Var "z"], Fun "g"[Var "a"] ]))))
+                       (T (Exists "x" (Conj (EqAtom (Fun "f"[Var "x", Var "z"]) (Var "y")) (PredAtom "p" [Fun "f"[Var "c", Var "z"], Fun "h"[Var "a"] ]))))
+                        zipperExample
+                    ) 
+                    (Ok (T (Exists "x" (Conj (EqAtom (Var "[]") (Var "y")) (PredAtom "p" [Fun "f"[Var "c", Var "z"], Fun "g"[Var "a"] ]))))) 
+            )
+        ]
```

Comment:
There should also be cases when `replaceTermWithVar` yields `Err`.

Reply:
I mean, you should create cases that will produce all kinds of errors that may occur – incompatible formulas, unsubstituable terms, formulas where replacement would be possible, but its does not match the provided equality.